### ISSUE TITLE
httplog gem not compatible, raising Algolia::AlgoliaProtocolError on request

### ISSUE THREAD

**Issue by pimpin** [algolia/algoliasearch-rails#273]
Hi there, we are currently needing httplog in our dev env but that seems not supported by algoliasearch-rails.

`Algolia::AlgoliaProtocolError : Cannot reach any host: not in gzip format, not in gzip format, not in gzip format, not in gzip format` is raised [here](https://github.com/algolia/algoliasearch-client-ruby/blob/1.12.7/lib/algolia/client.rb#L365)

Our app is a Rails 3 currently using
- algoliasearch-rails 1.19.1
- algoliasearch 1.12.7
- httplog 0.99.7
- httpclient 2.8.3

This might be only an issue for algoliasearch gem.
Is there any chance to solve the problem by updating algoliasearch-rails ?
I'll might check soon but any help will be appreciate.

**Comment by pimpin** [comment]
I've updated my gems but the bug is still there.
![image](https://user-images.githubusercontent.com/462084/32617555-8d890380-c575-11e7-8a72-ff2c6c2a2e05.png)
Could someone from Algolia ruby team give me an hand please ?

**Comment by redox** [comment]
Oh wow, that's weird @pimpin :/ Where do you deploy that code? Any proxy or so that could ultimately answer the "non" gziped answer?

**Comment by shingara** [comment]
this issue is only in our dev environment in a docker container. we request algolia directly

**Comment by pimpin** [comment]
@redox I've found that bug is gone when httplog is configured for [compact logging](https://github.com/trusche/httplog#compact-logging).
But data and headers are not logged in this compact log and we do want to keep them.

**Comment by redox** [comment]
> @redox I've found that bug is gone when httplog is configured for compact logging.
> But data and headers are not logged in this compact log and we do want to keep them.

Wow really?! Do you know if there is something we could do with the underlying HTTP API client to avoid that?

**Comment by shingara** [comment]
@pimpin it's not a valid solution when the goal is to have this log.

But seems a gzip issue so try with zclient.transparent_gzip_decompression = false`

https://github.com/algolia/algoliasearch-client-ruby/blob/1.12.7/lib/algolia/client.rb#L392

I think you can have more information about what is the really request or answer return by algolia this will avoid `not a gzip exception`

**Comment by pimpin** [comment]
@shingara I agree, it's not a solution to use the httplog compact option ; that's why I said
> we do want to keep them. [ie data and headers]

I've tried `client.transparent_gzip_decompression = false` and the bug is over !! Thank you.
Should we open a PR on [the client repo](https://github.com/algolia/algoliasearch-client-ruby) ?

**Comment by pimpin** [comment]
With next client release, you'll need to set `gzip` option to false as long as response is already decrompressed by any proxy or middleware like HttpLog gem.

```
if Rails.env.development?
  AlgoliaSearch.configuration = AlgoliaSearch.configuration.merge {gzip: false}
end
```

**Comment by Jerskouille** [comment]
Released w/ 1.18.2 of the ruby client @pimpin btw.

**Comment by pimpin** [comment]
Great. Thank you @Jerskouille
